### PR TITLE
docs: gomod artifacts hostRules

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Adds a description of how go artifacts updating constructs git env.

## Context

I think it's better here than in the private auth area.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/gomod/readme.md [hunk 1]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
+Before executing `go` commands to update the `go.sum`, Renovate will export `git` [`insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) directives in environment variables.
+
+The following logic is executed prior to "artifacts" updating:
+
+The token from the `hostRules` entry matching `hostType=github` and `matchHost=api.github.com` is added as the default authentication for `github.com`.
+For those running against `github.com`, this will be the default platform token.
+
+Next, all hostRules with both a token and matchHost will be fetched, except for any github.com one from above.
+
+Rules from this list will be converted to environment variable directives it they matchy any of the following characteristics:
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:39]:
```suggestion
Rules from this list are converted to environment variable directives if they match _any_ of the following characteristics:
```

File: lib/modules/manager/gomod/readme.md [hunk 2]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
+ - No hostType is defined, or
+ - hostType is `go`, or
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:41]:
```suggestion
 - `hostType` is `go`, or
```

File: lib/modules/manager/gomod/readme.md [hunk 3]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
+ - hostType is a platform (`github`, `gitlab`, `azure`, etc.)
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:42]:
```suggestion
 - `hostType` is a platform (`github`, `gitlab`, `azure`, etc.)
```

File: lib/modules/manager/gomod/readme.md [hunk 4]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
+
+The following logic is executed prior to "artifacts" updating:
+
+The token from the `hostRules` entry matching `hostType=github` and `matchHost=api.github.com` is added as the default authentication for `github.com`.
+For those running against `github.com`, this will be the default platform token.
+
+Next, all hostRules with both a token and matchHost will be fetched, except for any github.com one from above.
+
+Rules from this list will be converted to environment variable directives it they matchy any of the following characteristics:
+ - No hostType is defined, or
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:40]:
```suggestion
 - No `hostType` is defined, or
```

File: lib/modules/manager/gomod/readme.md [hunk 5]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
+### Private Modules Authentication
+
+Before executing `go` commands to update the `go.sum`, Renovate will export `git` [`insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) directives in environment variables.
+
+The following logic is executed prior to "artifacts" updating:
+
+The token from the `hostRules` entry matching `hostType=github` and `matchHost=api.github.com` is added as the default authentication for `github.com`.
+For those running against `github.com`, this will be the default platform token.
+
+Next, all hostRules with both a token and matchHost will be fetched, except for any github.com one from above.
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:37]:
```suggestion
Next, all `hostRules` with both a token and `matchHost` will be fetched, except for any github.com one from above.
```

File: lib/modules/manager/gomod/readme.md [hunk 6]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
   ]
 }
 ```
+
+### Private Modules Authentication
+
+Before executing `go` commands to update the `go.sum`, Renovate will export `git` [`insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) directives in environment variables.
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:30]:
```suggestion
Before running the `go` commands to update the `go.sum`, Renovate exports `git` [`insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) directives in environment variables.
```

File: lib/modules/manager/gomod/readme.md [hunk 7]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
+
+The following logic is executed prior to "artifacts" updating:
+
+The token from the `hostRules` entry matching `hostType=github` and `matchHost=api.github.com` is added as the default authentication for `github.com`.
+For those running against `github.com`, this will be the default platform token.
```
Comment (HonkingGoose) [inline lib/modules/manager/gomod/readme.md:35]: For me it's not clear at a glance what _this_ points at. Maybe you can explain what _this_ is?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/gomod/readme.md [hunk 8]
```diff
@@ -22,3 +24,19 @@ Indirect updates are disabled by default. To enable them, add a package rule suc
...
 ```
+
+### Private Modules Authentication
+
+Before running the `go` commands to update the `go.sum`, Renovate exports `git` [`insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) directives in environment variables.
+
+The following logic is executed prior to "artifacts" updating:
+
+The token from the `hostRules` entry matching `hostType=github` and `matchHost=api.github.com` is added as the default authentication for `github.com`.
+For those running against `github.com`, this will be the default platform token.
```
Comment (rarkins) [inline lib/modules/manager/gomod/readme.md:35]:
```suggestion
For those running against `github.com`, this token will be the default platform token.
```

**Pull request merged**